### DISCUSSION TITLE
Project-wide diagnostics: How to make it work also on not opened tabs?

### DISCUSSION THREAD

**Discussion by nerder** [zed-industries/zed#16805] [Feature Requests]
As in the question. My expectation is that project-wide diagnostics works for all project as in "I don't need to open the file" to find out that there is an error in it.

Is this something I need to manually configure? (NOTE: i'm using TS)

**Comment by jasonkuhrt** [comment]
For TS users, may be of help: https://github.com/yioneko/vtsls/blob/main/packages/service/configuration.schema.json#L1090

Example:

```jsonc
// Also possible to try: https://github.com/typescript-language-server/typescript-language-server
    // @see https://github.com/yioneko/vtsls
    "vtsls": {
      // @see https://github.com/yioneko/vtsls/blob/main/packages/service/configuration.schema.json
      "settings": {
        "typescript": {
          "tsserver": {
            "experimental": {
              "enableProjectDiagnostics": true
            }
          },
          "updateImportsOnFileMove": {
            "enabled": "always"
          }
        }
      }
    }
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
